### PR TITLE
CLN: Deprecate Index.summary (GH18217)

### LINKED ISSUE #18217: Deprecate Index.summary
```python
In [88]: idx = pd.Index([1, 2, 3])

In [89]: idx.summary()
Out[89]: 'Int64Index: 3 entries, 1 to 3'
```

### PR DESCRIPTION
**Pull request by GGordonGordon**
- [X] closes #18217 
- [X] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2185,7 +2185,7 @@ def __getitem__(self, key):
         try:
             if key in self.columns and not is_mi_columns:
                 return self._getitem_column(key)
-        except:
+        except Exception:
```
Comment (jreback) [inline pandas/core/frame.py]: either leave these  (IOW the linter is not actually failing these on CI), or catch a specific exception

File: pandas/core/indexes/datetimelike.py [hunk 2]
```diff
@@ -963,6 +963,15 @@ def summary(self, name=None):
         result = result.replace("'", "")
         return result
 
+    def summary(self, name=None):
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py]: you only need to deprecate in base class one (the sub-class is the same)

File: pandas/core/indexes/datetimelike.py [hunk 3]
```diff
@@ -941,7 +941,7 @@ def where(self, cond, other=None):
         return self._shallow_copy(result,
                                   **self._get_attributes_dict())
 
-    def summary(self, name=None):
+    def _summary(self, name=None):
         """
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py]: can you update the doc-strings here (and in the base class)

File: pandas/tests/indexes/timedeltas/test_formats.py [hunk 4]
```diff
@@ -67,6 +69,11 @@ def test_representation_to_series(self):
                 result = repr(pd.Series(idx))
                 assert result == expected
 
+    def test_summary_deprecated(self):
```
Comment (jreback) [inline pandas/tests/indexes/timedeltas/test_formats.py]: this is not necessary

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you rebase and update according to comments.

File: pandas/core/indexes/base.py [hunk 5]
```diff
@@ -1385,7 +1385,12 @@ def _has_complex_internals(self):
         # to disable groupby tricks in MultiIndex
         return False
 
-    def summary(self, name=None):
+    def _summary(self, name=None):
+        """
+        Return a summarized representation
```
Comment (jreback) [inline pandas/core/indexes/base.py]: you don't need a deprecated tag on the ``_summary`` methods

File: pandas/core/indexes/base.py [hunk 6]
```diff
@@ -1404,6 +1409,16 @@ def summary(self, name=None):
             name = type(self).__name__
         return '%s: %s entries%s' % (name, len(self), index_summary)
 
+    def summary(self, name=None):
+        """
+        Return a summarized representation
```
Comment (jreback) [inline pandas/core/indexes/base.py]: can you add a Parameters & Returns section

File: pandas/core/indexes/datetimelike.py (hunk 3 above)
Comment (jreback) [reply pandas/core/indexes/datetimelike.py]: don't need deprecated in the _summary doc-strings

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py [hunk 7]
```diff
@@ -1404,6 +1416,16 @@ def summary(self, name=None):
...
+        .. deprecated:: 0.23.0
+           No longer supported
```
Comment (jreback) [inline pandas/core/indexes/base.py]: you don't need the 'No longer supported'

**Comment by jreback** [comment]
minor comment. ping when pushed.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 7 above)
Comment (GGordonGordon) [reply pandas/core/indexes/base.py]: @jreback - This has been removed per your request. Update has been pushed.

**Review by jorisvandenbossche — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @GGordonGordon

**Pull request merged**